rustls/Cargo.toml

```diff
@@ -26,6 +26,7 @@ subtle = { version = "2.5.0", default-features = false }
 webpki = { package = "rustls-webpki", version = "0.102.4", features = ["alloc"], default-features = false }
 pki-types = { package = "rustls-pki-types", version = "1.7", features = ["alloc"] }
 zeroize = "1.7"
+zlib-rs = { version = "0.1", optional = true }
```

Comment:
I think it would be nice to front this with an explicitly named feature `zlib = ["zlib-rs"]` like we do for brotli. This is generally the better, more flexible approach (AIUI implicit features might be going away in 2024 edition).

Reply:
Should the feature gates after this change use `zlib-rs` or `zlib`? Both work, but `zlib-rs` seems more precise but less clear.

Reply:
I think that depends on whether we want the feature to be about the precise dependency (`zlib-rs`) vs about the capability (`zlib`). Do you think we'll ever want to have first-party support for multiple implementations of the same algorithm?

Reply:
I think it's more about the capability -- if (say) there was a better option for zlib we'd probably move to that rather than supporting both.